make_and_run_binder.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import sys
+import shutil
+import subprocess
+import git
+import re
+from distutils.sysconfig import get_python_inc
+
+
+# Overall script settings
+bindings_dir = 'cmake_bindings'
+this_project_source = f'{os.getcwd()}/src'
+this_project_include = f'{os.getcwd()}/include' 
+this_project_namespace_to_bind = 'bdsg'
+python_module_name = 'bdsg'
+
+def clone_repos():
+    # download the most recent copy of binder from git
+    if not glob.glob("binder"):
+        print("Binder not found, cloning repo...")
+        os.mkdir("binder")
+        git.Git(".").clone("https://github.com/RosettaCommons/binder.git")
+
+def build_binder():
+    # check for binder executable in the location we expect it
+    # if it's not there, build binder with the included script
+    # returns location of executable
+    if not glob.glob("./build/*/*/bin/*"):
+        print("Binder not compiled, using packaged build.py...")
+        os.system(f'{get_python_inc().split("/")[-1]} build.py')
+    pybind_source = f'binder/build/pybind11/include'
+    return "binder/" + glob.glob('./build/*/*/bin/')[0] + "binder"
+
+def clean_includes():
+    # goes through source code and replaces all quote-format includes with
+    # carrot-style includes.
+    # returns a dict with the files that were changed and what changes were made
+    changes_made = dict()
+    matcher = re.compile('^\s*#include "')
+    # find instances of includes we need to change
+    for filename in (glob.glob(f'{this_project_source}/**/*.hpp', recursive=True) + 
+                     glob.glob(f'{this_project_source}/**/*.cpp', recursive=True) +
+                     glob.glob(f'{this_project_source}/**/*.h', recursive=True) +
+                     glob.glob(f'{this_project_source}/**/*.cc', recursive=True) + 
+                     glob.glob(f'{this_project_source}/**/*.c', recursive=True) + 
+                     glob.glob(f'{this_project_source}/../include/**/*.hpp', recursive=True) + # this format needed to reach headers
+                     glob.glob(f'{this_project_source}/../include/**/*.cpp', recursive=True) +
+                     glob.glob(f'{this_project_source}/../include/**/*.h', recursive=True) +
+                     glob.glob(f'{this_project_source}/../include/**/*.cc', recursive=True) +
+                     glob.glob(f'{this_project_source}/../include/**/*.c', recursive=True) +
+                     glob.glob(f'{this_project_source}/../build/*/src/*/*.hpp', recursive=True) + 
+                     glob.glob(f'{this_project_source}/../build/*/src/*/*.cpp', recursive=True) + 
+                     glob.glob(f'{this_project_source}/../build/*/src/*/*.h', recursive=True) + 
+                     glob.glob(f'{this_project_source}/../build/*/src/*/*.cc', recursive=True) + 
+                     glob.glob(f'{this_project_source}/../build/*/src/*/*.c', recursive=True)):
```

Comment:
I'm sure this could be condensed to a collection of extensions and maybe `os.walk`.

Reply:
I didn't know this was an option-- I'll look into it.